### PR TITLE
Test for cache `wasm-bindgen` binary, not directory

### PR DESCRIPTION
**Pull request by danwilhelm**
Issue #508

1. Tests the cache for the `wasm-bindgen` binary. Previously, only
tested for the directory.

2. Removes the `wasm-bindgen` cache directory if it already exists,
prior to renaming the temporary cache directory.

Verified fix works correctly via the reproduction steps in #508 on both Windows and Mac.

- [✅] You have the latest version of `rustfmt` installed
- [✅] You ran `cargo fmt` on the code base before submitting
- [✅] You reference which issue is being closed in the PR text

### CODE REVIEW

File: src/bindgen.rs [hunk 1]
```diff
@@ -109,13 +109,15 @@ pub fn cargo_install_wasm_bindgen(
 
     let dirname = format!("wasm-bindgen-cargo-install-{}", version);
     let destination = cache.join(dirname.as_ref());
-    if destination.exists() {
+    let destination_dl = Download::at(&destination);
+
+    if let Ok(bindgen_path) = destination_dl.binary("wasm-bindgen") {
```
Comment (fitzgen) [inline src/bindgen.rs]: Nice.

File: src/bindgen.rs [hunk 2]
```diff
@@ -163,10 +165,21 @@ pub fn cargo_install_wasm_bindgen(
         })?;
     }
 
+    // Remove destination directory if already exists,
+    // e.g. if only the bindgen binary does not exist.
+    if destination.exists() {
+        fs::remove_dir_all(&destination).with_context(|_| {
```
Comment (fitzgen) [inline src/bindgen.rs]: Is this necessary? The rename doesn't overwrite the old directory anyways?

**Comment by alexcrichton** [comment]
I'm not sure I fully understand the bug here, can you help explain it a bit more? The intention was that if this directory exists then that signifies a successful installation, so this may mean that something with installation is awry or setting something up too soon perhaps?

**Comment by ashleygwilliams** [comment]
closing in favor of https://github.com/rustwasm/wasm-pack/pull/510 which is gonna include some refactors and unit tests

**Pull request closed without merging**